zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/NotebookRepoSync.java

```diff
@@ -55,14 +59,27 @@
 
   private List<NotebookRepo> repos = new ArrayList<>();
   private final boolean oneWaySync;
+  
+  private String notePersistence;
+  private static final String GLOBAL_SETTINGS_NAME = "Global Settings";
 
   /**
+   * Options for persisting notes on actions.
+   */
+  public enum NotePersist {
+    CONTINUOUS,       // Default value when all edit actions initiate note persist
+    RUN,              // Persist notes on run and checkpoint (commit) actions 
+    CHECKPOINT        // Persist note on checkpoint (commit) actions only
+  }
```

Comment:
it applies to base class `NotebookRepoSync` and consecutively to all repo instances when saving. instead of duplicating it in every repo, it's better to keep it here as global setting